typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts

```diff
@@ -153,14 +165,87 @@ export default class Axes2DLayer extends Layer<Axes2DLayerProps> {
     }
 
     updateState() {
-        const fontTexture = this.state["fontTexture"];
-        const { label_models, line_model, background_model } = this._getModels(
-            fontTexture as UniformValue
+        // Calculating margins.
+
+        // Note due to vertical scaling pixel2world mapping may dffer in X and Y direction.
+        const m = 100; // Length in pixels
+        let worldFrom = this.context.viewport.unproject([0, 0, 0]);
+        let worldTo = this.context.viewport.unproject([m, 0, 0]);
+        let v = [
+            worldFrom[0] - worldTo[0],
+            worldFrom[1] - worldTo[1],
+            worldFrom[2] - worldTo[2],
+        ];
+        const pixel2worldHor = Math.sqrt(v[0] * v[0] + v[1] * v[1]) / m;
+
+        worldFrom = this.context.viewport.unproject([0, 0, 0]);
+        worldTo = this.context.viewport.unproject([0, m, 0]);
+        v = [
+            worldFrom[0] - worldTo[0],
+            worldFrom[1] - worldTo[1],
+            worldFrom[2] - worldTo[2],
+        ];
+        const pixel2worldVer = Math.sqrt(v[0] * v[0] + v[1] * v[1]) / m;
+
+        const marginV = this.props.minimalMarginV ?? this.props.marginV;
+        const marginH = this.props.minimalMarginH ?? this.props.marginH;
+        let mv = marginV * pixel2worldVer;
+        let mh = marginH * pixel2worldHor;
+
+        // If specified horisontal margin (mh) is to small for the label, increase it.
+        const fontSizePixels = GetPixelsScale(
+            this.props.labelFontSizePt ?? defaultProps.labelFontSizePt
+        );
+        const viewportBoundsW = this.context.viewport.getBounds(); //bounds in world coordinates.
+        const yBoundsMin = viewportBoundsW[1];
+        const yBoundsMax = viewportBoundsW[3];
+        const isB = this.props.isBottomRuler;
+        const isT = this.props.isTopRuler;
+        const ymin = isB ? yBoundsMin + mv : yBoundsMin;
+        const ymax = isT ? yBoundsMax - mv : yBoundsMax;
+
+        const noLettersV = 1;
```

Comment:
noLettersV vs. nPixelsV below. Please be consistent.
Use numLettersV and numPixelsV ("no" is misleading, "n" too vague).